packages/next/server/render.tsx

```diff
@@ -571,7 +571,11 @@ export async function renderToHTML(
   }
 
   // url will always be set
-  const routerIsReady = !!(getServerSideProps || hasPageGetInitialProps)
+  const routerIsReady = !!(
+    getServerSideProps ||
+    hasPageGetInitialProps ||
+    !defaultAppGetInitialProps
```

Comment:
```suggestion
    (!defaultAppGetInitialProps && !isSSG)
```
A page with `getStaticProps` isn't considered `ready` until we're on the client and can determine if a query is present or not even with `getInitialProps` in `_app`. See [here](https://github.com/vercel/next.js/blob/622f63867a6fbd8fc8d84e782b66101f8c93bc65/packages/next/client/index.tsx#L185-L205) for related query updating that makes the `getStaticProps` page `ready`.

Reply:
Thanks! I applied the change in 5cc4ff8.